crates/relayer/src/chain/cosmos.rs

```diff
@@ -296,27 +296,27 @@ impl CosmosSdkChain {
             ));
         }
 
-        // Query /genesis RPC endpoint to retrieve the `max_expected_time_per_block` value to use as `max_block_time`.
-        // If it is not found, keep the configured `max_block_time`.
-        match self.block_on(self.rpc_client.genesis::<GenesisAppState>()) {
-            Ok(genesis_reponse) => {
-                let old_max_block_time = self.config.max_block_time;
-                let new_max_block_time =
-                    Duration::from_nanos(genesis_reponse.app_state.max_expected_time_per_block());
-
-                if old_max_block_time.as_secs() != new_max_block_time.as_secs() {
-                    self.config.max_block_time = new_max_block_time;
-
-                    info!(
-                        "Updated `max_block_time` using /genesis endpoint. Old value: `{}s`, new value: `{}s`",
-                        old_max_block_time.as_secs(),
+        // Query Connection Params with gRPC endpoint to retrieve the `max_expected_time_per_block` value and verify the
+        // configured `max_block_time`.
+        // If it is not found, the verification for the configured `max_block_time` is skipped.
```

Comment:
I think what we are missing is this case:
- user does not configure `max_block_time` in `config.toml`
- the UX should be that if chain can be queried we use  the chain `max_expected_time_per_block`
  - otherwise we use a default (30sec)
  
With this PR, if user does not configure it, the default is automatically loaded in the chain config and  if different we just see a warning, so cannot initialize here.
Before this PR the config was overriden with the genesis result (if avail). Not ideal UX either. 
Maybe to keep the same "bad" UX as before, as we cannot make it better, we override with the query value (with the warning that we do this). And we can figure out how to fix this in a different PR.

In my previous work, we used to have the notion of `configuration` vs `operational` values, kept separate. A tag can show `None` in configuration while showing some default (or other, e.g gRPC output in our case) value in`operational`. This way one can also make a difference between an unconfigured tag vs one configured with the default value (these requiring different UX handling)

Reply:
I see, I will setup the override using the queried value. This way we would have:

* If the `max_block_time` is not configured (the line is missing), it will use the queried value or the default `30s` if the query fails
* If the `max_block_time` is configured, it will use the queried value or the configured value if the query fails

Doesn't this fit the missing case? The difference is that instead of having an explicit `None`, the `max_block_time` is simply not set.

Reply:
Updated here ae81050a96c831116d930367b9356e80cb355b48

Reply:
> If the max_block_time is configured, it will use the queried value or the configured value if the query fails
> Doesn't this fit the missing case?

imo this last bullet should be:

- If the `max_block_time` is configured, the configured value is used. 

We could still query and issue a warning if the queried value is different but not override. But let's leave it like you have it for now.

Reply:
Oh I understand. We could do this by using an `Option`. I will create a follow-up issue for this

Reply:
https://github.com/informalsystems/hermes/issues/4147